public/actions.js

```diff
@@ -418,11 +418,18 @@ export const MoveMob = (state, { from, to }) => {
 };
 
 export const AddGoal = (state) => {
-  const goals = state.goals.concat({
-    id: Math.random().toString(36).slice(2),
-    text: state.goal,
-    completed: false,
-  });
+  var newGoals = state.goal.split(";");
+  var goals = state.goals;
+  for (let i = 0; i < newGoals.length; i++) {
+    var newGoal = newGoals[i].trim();
+    if (newGoal.length > 0) {
+      goals = goals.concat({
+        id: Math.random().toString(36).slice(2),
+        text: newGoal,
+        completed: false,
+      });
+    }
+  }
```

Comment:
A couple of things I think we could do to make this a little nicer.

First, let's make a new action, `export const AddMultipleGoals = (state, goals)`

Second, I think we can trim down the code using a map, which would look vaguely like this:

```js
const goals = state.goals.split(';').map((text) => {
  // ...
  return { ...objectThatLooksLikeAGoal };
});
```

Third, we'll need to test this new action. I can help you out there once you get here.

Fourth, wire it up to the UI.

How does that sound?

Reply:
Sounds great!